File: IOS/TMStore/TMStore/Classes/Services/TMShippingEngine/ShippingEngine.m
//
//  ShippingEngine.m
//  TMShippingSDK
//
//  Created by Rishabh Jain on 28/09/16.
//  Copyright © 2016 Twist Mobile. All rights reserved.
//

#import "ShippingEngine.h"
@implementation ShippingEngine
static BOOL listCities = false;
- (BOOL)hasSubDistrictSelection {
    return subDistrictSelection;
}
- (BOOL)hasDistrictSelection {
    return districtSelection;
}
- (BOOL)hasCitySelection {
    return citySelection;
}
- (BOOL)hasCountrySelection {
    return countrySelection;
}
- (BOOL)hasStateSelection {
    return stateSelection;
}
+ (BOOL)listCities {
    return listCities;
}
+ (BOOL)areCitiesListed {
    return listCities;
}
- (void)setListCities:(BOOL)value {
    listCities = value;
}
@end

File: Android/_setup/_ProjectName_/src/main/java/com/twist/dataengine/ShippingEngine.java
package com.twist.dataengine;

import com.twist.dataengine.entities.TM_Region;
import com.twist.dataengine.entities.TM_StoreInfo;

/**
 * Created by Twist Mobile on 8/18/2016.
 */

public abstract class ShippingEngine {

    public String request_url_countries = "";
    public String request_url_states = "";
    public String request_url_districts = "";
    public String request_url_sub_discricts = "";
    public String request_url_cities = "";
    public String request_url_calculate_shipping = "";
    public String request_url_find_shippings = "";
    public String request_url_currency = "";
    public String request_store_destination = "";

    protected boolean citySelection = false;
    protected boolean districtSelection = false;

    public boolean hasSubDistrictSelection() {
        return subDistrictSelection;
    }

    public boolean hasDistrictSelection() {
        return districtSelection;
    }

    public boolean hasCitySelection() {
        return citySelection;
    }

    protected boolean subDistrictSelection = false;

    public static boolean areCitiesListed() {
        return listCities;
    }

    protected static boolean listCities = false;

    public abstract void getCountries(TM_Region parent, DataQueryHandler dataQueryHandler);

    public abstract void getStates(TM_Region parent, DataQueryHandler dataQueryHandler);

    public abstract void getDistricts(TM_Region parent, DataQueryHandler dataQueryHandler);

    public abstract void getSubDistricts(TM_Region parent, DataQueryHandler dataQueryHandler);

    public abstract void getSubDistricts(TM_Region parent,String state, DataQueryHandler dataQueryHandler);

    public abstract void getCities(TM_Region parent, DataQueryHandler dataQueryHandler);

    public abstract void calculateShipping(TM_Region origin, TM_Region destination, DataQueryHandler dataQueryHandler);

    public abstract void getAvailableShipping(TM_StoreInfo origin, TM_Region destination, float weight, DataQueryHandler dataQueryHandler);

    //public abstract void getAvailableShipping(String city,String state, DataQueryHandler dataQueryHandler);

    public abstract void getCurrencyRate(DataQueryHandler dataQueryHandler);

    public abstract void getStoreLocation(final DataQueryHandler dataQueryHandler);
}
